proxy/src/telemetry/sensor/http.rs

```diff
@@ -222,19 +222,25 @@ where
     }
 
     fn call(&mut self, mut req: Self::Request) -> Self::Future {
-        let (inner, body_inner) = match req.extensions_mut().remove::<Arc<ctx::transport::Server>>() {
-            None => (None, None),
-            Some(ctx) => {
+        let metadata = (
+            req.extensions_mut().remove::<Arc<ctx::transport::Server>>(),
+            req.extensions_mut().remove::<RequestOpen>()
+        );
+        let (inner, body_inner) = match metadata {
+            (ctx @ None, request_open) | (ctx, request_open @ None) => {
+                warn!(
+                    "missing metadata for a request to {:?}; ctx={:?}; request_open={:?};",
+                    req.uri(), ctx, request_open
```

Comment:
Note that we log only the request URI here, because logging the entire request struct would require a `A: Debug` bound on the request's body type, which I didn't want to add.